modules/dlm-lifecycle-iam-role/main.tf

```diff
@@ -0,0 +1,56 @@
+variable "iam_role_name" {
+  description = "The IAM role name for the DLM lifecyle policy"
+  type        = string
+  default     = "dlm-lifecycle-role"
+}
+
+# Create the iam role
+resource "aws_iam_role" "dlm_lifecycle_role" {
+  name                = var.iam_role_name
+  assume_role_policy  = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "dlm.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
```

Comment:
We can merge this as-is, but we should update this code to use the data source so we can use HCL and don't have to do the messy JSON thing.